autogpt/scripts/llamafile/serve.py

```diff
@@ -30,36 +38,64 @@ def report_download_progress(chunk_number: int, chunk_size: int, total_size: int
         )
 
 
-def download_llamafile():
-    print(f"Downloading {LLAMAFILE.name}...")
-    import urllib.request
+if __name__ == "__main__":
+    # Go to autogpt/scripts/llamafile/
+    os.chdir(Path(__file__).resolve().parent)
 
-    url = "https://huggingface.co/jartine/Mistral-7B-Instruct-v0.2-llamafile/resolve/main/mistral-7b-instruct-v0.2.Q5_K_M.llamafile"  # noqa
+    if not LLAMAFILE.is_file():
+        download_file(LLAMAFILE_URL, LLAMAFILE)
 
-    urllib.request.urlretrieve(url, LLAMAFILE.name, reporthook=report_download_progress)
-    print()
+        if platform.system() != "Windows":
+            LLAMAFILE.chmod(0o755)
+            subprocess.run([LLAMAFILE, "--version"], check=True)
 
-    LLAMAFILE.chmod(0o755)
-    subprocess.run([LLAMAFILE, "--version"], check=True)
+        print(
+            "\n"
+            "NOTE: To use other models besides mistral-7b-instruct-v0.2, "
+            "download them into autogpt/scripts/llamafile/"
+        )
 
-    print(
-        "\n"
-        "NOTE: To use other models besides mistral-7b-instruct-v0.2, "
-        "download them into autogpt/scripts/llamafile/"
+    if platform.system() != "Windows":
+        base_command = [LLAMAFILE]
+    else:
+        # Windows does not allow executables over 4GB, so we have to download a
+        # model-less llamafile.exe and extract the model weights (.gguf file)
```

Comment:
We don’t have to do that just run the Llamafile exe and pass it the other model

Reply:
oo